### DISCUSSION TITLE
Intergration with Syndikate (Syndi Kate)

### DISCUSSION THREAD

**Discussion by sjehuda** [lwindolf/liferea#1393] [General]
Greetings!

Kate is the woman who makes sure that you and your children would be informed with the news that you want.

She is determined that you will be in control of the data that you want.

She has a strong desire for people to enjoy martimonial and peaceful family life, and she strongly believes that one of the important factors to accomlish this, is that you must be in control of the data that is being delivered to you. 

Did we mention that Kate is a strong believer of independence?

Kate's favourite telecmmunication protocols are HTTP and XMPP. You can find her on IRC too.

Kate is a mother to five children and maintains several farms across Austria, Germany and Switzerland!

At her leisure time, she reads books, and listens to classical music, and during the weekends Kate rides horses with her husband and their eldest daughter.

The family's favourite travel hubs are the Bavarian village Mountaineers in Kreuth (the home town of Kate), and during the summer, the family often resides in Soizkaumaguad and travels across the wounderful Lake Wolfgang of Austria while they often visit her husband's family in his home town, the city of Salzburg.

Viel Glück und Viel Segen!
Syndikate

---

### See https://github.com/lwindolf/liferea/discussions/974#discussioncomment-8195898

<details>

I did not yet publish the project, and yet...

![image](https://github.com/user-attachments/assets/4e73bcdc-b32b-41ad-9ca2-a9957895a0ed)

![image](https://github.com/user-attachments/assets/a5dad2b6-7c0f-43d8-8c49-12a5cbe53dce)

![image](https://github.com/user-attachments/assets/9fa4bc4d-a21c-44e5-afa6-a7623df3b2a2)

![image](https://github.com/user-attachments/assets/c582fe88-6a65-44ad-92f5-262653e27feb)

Kate is expected to be arriving, very soon...

</details>

**Comment by sjehuda** [comment]
@lwindolf

An initial database is ready for destribution.

I need to build an HTTP API, and I will utilize FastAPI for that purpose.

Do you have any special advise of a recommended manner to convey HTTP queries to server?

**Comment by sjehuda** [comment]
Related https://github.com/lwindolf/liferea/issues/967 https://github.com/lwindolf/liferea/issues/1371

**Reply by lwindolf** [reply]
@sjehuda Would you be interested on integrating Syndikate in https://lwindolf.github.io/rss-finder/ ?

**Reply by sjehuda** [reply]
Yes. I would.

Generally, the Syndikate project is only a couple of components, a
database and an API (query parameters over HTTP), though I intend to add
support for XMPP as well.

Syndikate is based upon FastAPI (Python framework which is similar to
Flask).

It will have a few pages. Essentials are marked with *

1. The main page. *
2. List news software. *
3. Campaign for syndicated content.
4. Maybe other pages (statistics etc.)
5. Admin page to control/censor unwanted sources by some premises.

I think, that the actual concern is to share common structures for
database and query parameters, so such subscription sharing system
would be interoperable over any instance.

It is really a small project, and a database.

I have sent to you an email in order to negotiate the sharing of other
details.

**Reply by sjehuda** [reply]
Lars.

I suppose that /rss-finder/ is a portal.

Syndikate will be released as an open source software, so I guess that you can establish your own instance, as I do not intend to do so yet.

I am currently working on adding support for Gemini and twtxt, and I intend to do some other code "clean up" https://git.xmpp-it.net/sch/Slixfeed

However, if you deem it urgent (please feel free to demand however you desire), then I will strive to publish a working test version of Syndikate so that would be provided within the next couple of weeks.

**Reply by lwindolf** [reply]
Hmmm... from my side I can exclude hosting any piece of software I need to run on a server. I plan to have rss-finder as a PWA that can be run from every user using only client side processing.

**Reply by sjehuda** [reply]
The SQLite database file currently contains 5,000 subscriptions, and weights less than 2MB, so it would also be possible to utilize it locally. What could be a preferable filetype for that database?

**Reply by lwindolf** [reply]
2MB sqlite could be used in Javascript using the Javascript version of the sqlite client. It does not even need to be converted.
